File: Sliding Window/leetcode_187.cpp

// Problem link ===>> https://leetcode.com/problems/repeated-dna-sequences/description/?envType=problem-list-v2&envId=21r1z5od

#include <bits/stdc++.h>
#include <ext/pb_ds/assoc_container.hpp>
#include <ext/pb_ds/tree_policy.hpp>
using namespace std;
using namespace __gnu_pbds;
#define ll long long
#define ld long double
#define endl '\n'
const int MOD = 1e9 + 7;
const double EPS = 1e-9;
const double PI = acos(-1.0);
template <typename T>
using ordered_set = tree<T, null_type, less<T>, rb_tree_tag, tree_order_statistics_node_update>;
template <typename T>
using ordered_set_mul = tree<T, null_type, less_equal<T>, rb_tree_tag, tree_order_statistics_node_update>;

class Solution
{
public:
    vector<string> findRepeatedDnaSequences(string s)
    {
        unordered_map<string, int> seen;
        vector<string> result;

        for (int i = 0; i + 9 < s.length(); i++)
        {
            string substr = s.substr(i, 10);
            seen[substr]++;
            if (seen[substr] == 2)
            {
                result.push_back(substr);
            }
        }

        return result;
    }
};

int main()
{
    ios_base::sync_with_stdio(false);
    cout.tie(nullptr);
    cin.tie(nullptr);
    
    ll t;
    cin >> t;
    while (t--)
    {
        ll n;
        cin >> n;

        
    }
    return 0;
}

File: Sliding Window/leetcode_187.py

# Problem link ===>> https://leetcode.com/problems/repeated-dna-sequences/description/?envType=problem-list-v2&envId=21r1z5od
# submission link == =>> https://leetcode.com/problems/repeated-dna-sequences/submissions/1695486516/?envType=problem-list-v2&envId=21r1z5od

import sys
import math
import bisect
import heapq
from collections import defaultdict, deque, OrderedDict, Counter
from functools import lru_cache, reduce
from itertools import permutations, combinations, accumulate
from typing import List, Tuple, Dict, Set, Optional, Union, Any

MOD = 10**9 + 7
INF = float('inf')
input = sys.stdin.read
print = sys.stdout.write


class Solution:
    def findRepeatedDnaSequences(self, s: str) -> List[str]:
        seen = {}
        result = []

        for i in range(len(s) - 9):
            substr = s[i:i+10]
            seen[substr] = seen.get(substr, 0) + 1
            if seen[substr] == 2:
                result.append(substr)

        return result

def main():
    t = int(input())
    for _ in range(t):
        n = int(input())

if __name__ == "__main__":
    solution = Solution()
